pandas/core/indexes/period.py

```diff
@@ -54,19 +53,49 @@ def f(self):
     f.__doc__ = fget.__doc__
     return property(f)
 
-
 # --- Period index sketch
 
 
 def _new_PeriodIndex(cls, **d):
     # GH13277 for unpickling
-    if d['data'].dtype == 'int64':
-        values = d.pop('data')
-    return cls._from_ordinals(values=values, **d)
+    values = d.pop('data')
+    if values.dtype == 'int64':
+        return cls._from_ordinals(values=values, **d)
+    else:
+        return cls(values, **d)
 
 
-class PeriodIndex(PeriodArrayMixin, DatelikeOps, DatetimeIndexOpsMixin,
-                  Int64Index):
+class PeriodDelegateMixin(PandasDelegate):
+    """
+    Delegate from PeriodIndex to PeriodArray.
+    """
+    def _delegate_property_get(self, name, *args, **kwargs):
+        result = getattr(self._data, name)
+        box_ops = (
+            set(PeriodArray._datetimelike_ops) - set(PeriodArray._bool_ops)
+        )
+        if name in box_ops:
+            result = Index(result, name=self.name)
+        return result
+
+    def _delegate_property_set(self, name, value, *args, **kwargs):
+        setattr(self._data, name, value)
+
+    def _delegate_method(self, name, *args, **kwargs):
+        result = operator.methodcaller(name, *args, **kwargs)(self._data)
+        return Index(result, name=self.name)
+
+
+@delegate_names(PeriodArray,
+                PeriodArray._datetimelike_ops + ['size', 'asi8', 'shape'],
```

Comment:
I think this delegation is a good idea for the dtype-specific ones, but it might be more explicit to keep general ones as `size` and `shape` as actual properties calling the underlying values (but could be shared for the different index classes based on EA). 
Anyway, not that important for this PR, this can be refactored after converting all Arrays.